### ISSUE TITLE
NoSuchMethodError: java.nio.ByteBuffer.flip()Ljava/nio/ByteBuffer

### ISSUE THREAD

**Issue by pan3793** [housepower/ClickHouse-Native-JDBC#242]
### Environment

- ClickHouse Native JDBC version: mainline code

### Error logs

```
# JMH version: 1.26
# VM version: JDK 1.8.0_272, OpenJDK 64-Bit Server VM, 25.272-b17
# VM invoker: /Users/chengpan/.sdkman/candidates/java/8.0.272.fx-zulu/zulu-8.jdk/Contents/Home/jre/bin/java
# VM options: <none>
# Warmup: <none>
# Measurement: 1 iterations, 10 s each
# Timeout: 10 min per iteration
# Threads: 1 thread, will synchronize iterations
# Benchmark mode: Average time, time/op
# Benchmark: com.github.housepower.jdbc.benchmark.InsertIBenchmark.benchInsertNative
# Parameters: (batchSize = 200000, columnNum = 50)

# Run progress: 8.33% complete, ETA 00:12:57
# Fork: 1 of 1
2020-12-08 23:08:37 [com.github.housepower.jdbc.benchmark.InsertIBenchmark.benchInsertNative-jmh-worker-1] INFO ClickHouseDriver - Driver registered
Iteration   1: <failure>

java.lang.NoSuchMethodError: java.nio.ByteBuffer.flip()Ljava/nio/ByteBuffer;
	at com.github.housepower.jdbc.data.ColumnWriterBuffer.writeTo(ColumnWriterBuffer.java:40)
	at com.github.housepower.jdbc.data.Column.flushToSerializer(Column.java:42)
	at com.github.housepower.jdbc.data.Block.writeTo(Block.java:122)
	at com.github.housepower.jdbc.protocol.DataRequest.writeImpl(DataRequest.java:49)
	at com.github.housepower.jdbc.protocol.Request.writeTo(Request.java:30)
	at com.github.housepower.jdbc.connect.NativeClient.sendRequest(NativeClient.java:160)
	at com.github.housepower.jdbc.connect.NativeClient.sendData(NativeClient.java:119)
	at com.github.housepower.jdbc.ClickHouseConnection.sendInsertRequest(ClickHouseConnection.java:202)
	at com.github.housepower.jdbc.statement.ClickHousePreparedInsertStatement.executeBatch(ClickHousePreparedInsertStatement.java:99)
	at com.github.housepower.jdbc.benchmark.InsertIBenchmark.lambda$new$0(InsertIBenchmark.java:54)
	at com.github.housepower.jdbc.benchmark.AbstractIBenchmark.withConnection(AbstractIBenchmark.java:80)
	at com.github.housepower.jdbc.benchmark.InsertIBenchmark.benchInsertNative(InsertIBenchmark.java:62)
	at com.github.housepower.jdbc.benchmark.jmh_generated.InsertIBenchmark_benchInsertNative_jmhTest.benchInsertNative_avgt_jmhStub(InsertIBenchmark_benchInsertNative_jmhTest.java:186)
	at com.github.housepower.jdbc.benchmark.jmh_generated.InsertIBenchmark_benchInsertNative_jmhTest.benchInsertNative_AverageTime(InsertIBenchmark_benchInsertNative_jmhTest.java:150)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:453)
	at org.openjdk.jmh.runner.BenchmarkHandler$BenchmarkTask.call(BenchmarkHandler.java:437)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

### Steps to reproduce

Run Benchmark

### Investigation

I found a similar issue report at https://bitbucket.org/ijabz/jaudiotagger/issues/313/java-8-javalangnosuchmethoderror 

TL;DR

It cause by JDK change, explicit upcast is necessary to avoid this issue.

```
ByteBuffer buffer = ...
// buffer.flip(); // use below code instead of this line
((Buffer) buffer).flip();
```
